code/procs/mobprocs/chatprocs.dm

```diff
@@ -1090,7 +1091,7 @@
 
 		if((isflockmob(M)) || (M.client.holder && !M.client.player_mode) || (isobserver(M) && !(istype(M, /mob/dead/target_observer/hivemind_observer))))
 			thisR = rendered
-		if(flock?.snooping && M.client && M.robot_talk_understand)
+		if(M.client && (M.robot_talk_understand || istype(M, /mob/living/intangible/aieye)))
```

Comment:
```suggestion
		if(M.robot_talk_understand || istype(M, /mob/living/intangible/aieye))
```
You're iterating through clients, so no need to check if it exists. Also is the `aieye` check necessary? The `robot_talk` proc doesn't have it.

Reply:
Yeah it is, not sure why

Reply:
I think the mainframe relays the robot talk to the AI eye so it doesn't have the var set? Could be worth looking into if setting the var on aieye doubles the messages or not.